_P082_GPS.py

```diff
@@ -7,312 +7,427 @@
-     self.set_value(2,self.lat,False)
-     self.set_value(3,self.GPSDAT['alt'],False)
-     self.set_value(4,self.GPSTRACK['speedKm'],False)
-    except:
-     pass
-    self.plugin_senddata()
-   self._lastdataservetime = rpieTime.millis()
-   result = True
-   self.readinprogress = 0
-  return result
-
- def connect(self):
-    try:
-     if self.serdev.isopened():
-      self.initialized = True
-      return True
-    except:
-     pass
-    try:
-     self.serdev.close() # close in case if already opened by ourself
-     self.serdev = None
-    except:
-     pass
-    try:
-     self.serdev = rpiSerial.SerialPort(self.taskdevicepluginconfig[0],self.baud,ptimeout=1,pbytesize=rpiSerial.EIGHTBITS,pstopbits=rpiSerial.STOPBITS_ONE)
-     misc.addLog(rpieGlobals.LOG_LEVEL_INFO,"Serial connected "+str(self.taskdevicepluginconfig[0]))
-    except Exception as e:
-     misc.addLog(rpieGlobals.LOG_LEVEL_ERROR,"Serial failed "+str(e))
-    try:
-     self.initialized = self.serdev.isopened()
-    except Exception as e:
-     self.initialized = False
-     misc.addLog(rpieGlobals.LOG_LEVEL_ERROR,"Open failed "+str(e))
-
- def bgreceiver(self):
-  if self.initialized:
-   self.validloc = -1
-   while self.enabled and self.initialized:
-    if self.serdev is not None:
-     try:
-      reading = self.serdev.readline()
-      if reading:
-       recdata = reading.decode("utf-8")
-       self.parseResponse(recdata)
-      else:
-       time.sleep(0.001)
-     except:
-      time.sleep(0.5)
-
-   try:
-     self.serdev.close()
-   except:
-     pass
-
- def parseResponse(self,gpsLine):
- #    global lastLocation
- #    gpsChars = ''.join(chr(c) for c in gpsLine)
- #    if "*" not in gpsChars:
- #        return False
-    gpsChars = str(gpsLine)
-    try:
-     if '*' in gpsChars:
-      gpsStr, chkSum = gpsChars.split('*')
-     elif '_' in gpsChars:
-      gpsStr, chkSum = gpsChars.split('_')
-     else:
-      chkSum = gpsChars[-2:]
-      gpsStr = gpsChars[:-2]
-    except:
-      chkSum = gpsChars[-2:]
-      gpsStr = gpsChars[:-2]
-    gpsComponents = gpsStr.split(',')
-    gpsStart = gpsComponents[0]
-#    print(gpsLine)#debug
-    if ("GGA" in gpsStart):
-#        print("Valid GGA from GPS") # DEBUG
-        if self.devfound==False:
-         misc.addLog(rpieGlobals.LOG_LEVEL_INFO,"GPS found")
-         self.devfound=True
-        prevval = self.validloc
-        self.validloc = 0
-        chkVal = 0
-        for ch in gpsStr[1:]: # Remove the $
-            chkVal ^= ord(ch)
-        if (chkVal == int(chkSum, 16)):
-            for i, k in enumerate(
-                ['strType', 'fixTime',
-                'lat', 'latDir', 'lon', 'lonDir',
-                'fixQual', 'numSat', 'horDil',
-                'alt', 'altUnit', 'galt', 'galtUnit',
-                'DPGS_updt', 'DPGS_ID']):
-                self.GPSDAT[k] = gpsComponents[i]
+    PLUGIN_ID = 82
+    PLUGIN_NAME = "Position - GNSS/GPS (TESTING)"
+    PLUGIN_VALUENAME1 = "Latitude"
```

Comment:
Does changing the order of variables like this render existing code (and rules) incompatible?